R/tokens.R

```diff
@@ -178,16 +178,67 @@ tokens.character <- function(x, ...) {
 #' @rdname tokens
 #' @export
 #' @noRd
-tokens.corpus <- function(x, ..., include_docvars = TRUE) {
+tokens.corpus <- function(x, ..., hash = TRUE, include_docvars = TRUE) {
     result <- tokens_internal(texts(x), ...)
     if (include_docvars) {
         docvars(result) <- documents(x)[, which(names(documents(x)) != "texts"), drop = FALSE]
     } else {
         docvars(result) <- data.frame(row.names = docnames(x))
     }
+    if (hash == FALSE)
+        result <- as.tokenizedTexts(result)
     return(result)
 }
 
+#' @rdname tokens
+#' @export
+#' @noRd
+#' 
+tokens.tokens <-  function(x, what = c("word", "sentence", "character", "fastestword", "fasterword"),
+                           remove_numbers = FALSE,
+                           remove_punct = FALSE,
+                           remove_symbols = FALSE,
+                           remove_separators = TRUE,
+                           remove_twitter = FALSE,
+                           remove_hyphens = FALSE,
+                           remove_url = FALSE,
+                           ngrams = 1L,
+                           skip = 0L,
+                           concatenator = "_",
+                           hash = TRUE,
+                           verbose = quanteda_options("verbose"),
+                           include_docvars,
+                           ...) {
+    
+    
+    if (remove_hyphens)
+        types(x) <- stri_replace_all_fixed(types(x), "-", " ")
+    if (remove_twitter)
+        types(x) <- stri_replace_all_fixed(types(x), c('@', '#'), "", vectorize_all = FALSE)
+    if (remove_hyphens || remove_twitter)
+        x <- tokens_recompile(x)
+    
+    regex <- c()
+    if (remove_numbers)
+        regex = c(regex, "^[\\p{N}]+$")
+    if (remove_punct)
+        regex = c(regex, "^[\\p{P}\\p{S}]+$")
+    if (remove_symbols)
+        regex = c(regex, "^[\\p{S}]+$")
+    if (remove_separators)
+        regex = c(regex, "^[\uFE00-\uFE0F\\p{Z}\\p{C}]+$") 
+    if (remove_url)
```

Comment:
with the exception of the `remove_url`, the others are char_class and might be faster implemented that way (through `stringi::stri_remove_charclass()` - although I realise that `tokens_remove()` does not currently provide this option. Something to note for the future? Or, we could examine the regex in `tokens_remove()` and if it matches a charclass, we call that instead of the equivalent regex function. We should test the performance differences before investing time in this.